### ISSUE TITLE
`ipfs.dag.put` does not derive `format` & `hashAlg` from the passed cid

### ISSUE THREAD

**Issue by Gozala** [ipfs/js-ipfs#3349]
I think there is an implementation bug in `ipfs.dag.put` which is supposed to derive `hashAlg` and `format` info from passed `options.cid` when it's provided, but it does not seem to do so:

https://github.com/ipfs/js-ipfs/blob/d27b6d1d138121e368b3bd3bc0c3ff6b4df8a29c/packages/ipfs-core/src/components/dag/put.js#L56-L66

In fact it doesn't seem to do anything with passed `cid`

https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-core/src/components/dag/put.js#L75-L79

Other than throw an error if it isn't provided (when `format` and `hashAlg` aren't provided either):

https://github.com/ipfs/js-ipfs/blob/d27b6d1d138121e368b3bd3bc0c3ff6b4df8a29c/packages/ipfs-core/src/components/dag/put.js#L75-L79

**Comment by mikeal** [comment]
why does the `dag.put()` even accept a CID. isn’t the whole point of this API to do the encoding for you? if you’ve already created a CID then you’ve already encoded the data, which means you could just use the block API right?

**Comment by achingbrain** [comment]
I don't think this API should accept a CID as an argument, please feel free to remove it.

**Comment by Gozala** [comment]
Sounds good, I'll remove `cid` option

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).

Please feel to reopen with any comments before 2023-06-05. We will do a final pass on reopened issues afterward (see https://github.com/ipfs/js-ipfs/issues/4336).
